ixnetwork/api/ixn_rest.py

```diff
@@ -75,28 +75,35 @@ def disconnect(self):
     #
 
     def waitForComplete(self, request, response, timeout=128):
-        if 'errors' in response.json():
-            raise TgnError(response.json()['errors'][0])
-        if response.json()['state'].lower() == 'error':
-            result = json.loads(response.content)['result'].strip()
-            raise TgnError(f'wait for post {response.url} failed - {result}')
-        if response.json()['state'].lower() == 'success':
-            return response
         progress_url = json.loads(response.content)[u'url']
+
         if 'http' not in progress_url:
             # sometimes the progress url is relative, for example in version 8.50 linux loadconfig
             progress_url = self.server_url + progress_url
+
         for _ in range(timeout):
-            response = self.get(progress_url)
-            if type(response.json()) == dict:
-                state = response.json()['state']
-            else:
-                state = response.json()[0]['state']
-            if state.lower() not in ['in_progress', 'down']:
+            response_data = response.json()
+            if type(response_data) is not dict:
+                response_data = response_data[0]
+
+            if 'errors' in response_data:
+                raise TgnError(response_data['errors'][0])
+
+            state = response_data.get('state', 'unknown').lower()
+
+            if state == 'error':
+                result = json.loads(response.content)['result'].strip()
+                raise TgnError(f'wait for post {response.url} failed - {result}')
+            elif state == 'success':
+                return response
+            elif state not in ['unknown', 'in_progress', 'down']:
                 return response
```

Comment:
What are the other states that you return but do not raise TgnError?
It seems that the callers are expecting this method be deterministic. Either the operation completed successfully or we raise an error.

Reply:
This logic was already there, I've just extended it to include the case where the 'state' attribute is missing (and I then default it to 'unknown').  Because I haven't tested this against different Ixia versions I didn't want to break any pre-existing functionality.

I would hazard a guess that we will always be given { 'state': 'SUCCESS' } from the API, but previously this is the test you were previously using inside the for-loop to check for success:

```python
            if state.lower() not in ['in_progress', 'down']:
                return response
```